src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java

```diff
@@ -57,6 +57,13 @@ public final class ThrowableToErrorMapper {
           if (message.contains("vector<float,")) {
             message = "Mismatched vector dimension";
           }
+          if (message.contains(
+                  "If you want to execute this query despite the performance unpredictability, use ALLOW FILTERING")
+              || message.contains("ANN ordering by vector requires the column to be indexed")) {
```

Comment:
Does this mean that any error message that contains "If you want to execute this query despite the performance unpredictability, use ALLOW FILTERING" will be replaced with "Faulty collection (missing indexes). Recommend re-creating the collection"?

I think the new error message may be missing some valuable information, like the fact that this error is due to vector search or a query on an unindexed field. Maybe we can add some more details, like some info on which index is missing? Or at least have separate error messages for vector search vs other missing indexes?

Reply:
I'm not sure that knowing which index was impacted has much benefit to the user, as it doesn't change the required action.

Reply:
As long as there is indication of missing index(es), makes sense to me.
Knowing which index(es) missing would be somewhat useful to us (maintainers) but not so much users, agree.